internal/nodes/avi_model_nodes.go

```diff
@@ -612,21 +619,24 @@ func (o *AviVsNode) ReplaceSniPGInSNINode(newPGNode *AviPoolGroupNode, key strin
 	}
 	// If we have reached here it means we haven't found a match. Just append.
 	o.PoolGroupRefs = append(o.PoolGroupRefs, newPGNode)
-	return
 }
 
-func (o *AviVsNode) ReplaceSniHTTPRefInSNINode(newHttpNode *AviHttpPolicySetNode, key string) {
+func (o *AviVsNode) ReplaceSniHTTPRefInSNINode(httpPGPath AviHostPathPortPoolPG, httpPolName, key string) {
 	for i, http := range o.HttpPolicyRefs {
-		if http.Name == newHttpNode.Name {
-			o.HttpPolicyRefs = append(o.HttpPolicyRefs[:i], o.HttpPolicyRefs[i+1:]...)
-			o.HttpPolicyRefs = append(o.HttpPolicyRefs, newHttpNode)
-			utils.AviLog.Infof("key: %s, msg: replaced sni http in model: %s Pool name: %s", key, o.Name, http.Name)
-			return
+		if http.Name == httpPolName {
+			for j, hppMap := range o.HttpPolicyRefs[i].HppMap {
+				if hppMap.Name == httpPGPath.Name {
```

Comment:
Can you explain what is going on here? It seems we are deleting the element from the slice twice?

Reply:
Each HTTP PS will have multiple entries of HPP map, corresponding to path.  We are replacing old hpp map entry with latest one with this code base.

Reply:
To explain HTTP PS structure for host foo.com with /foo and /bar in default ns ...
VS will have single HTTP PS with naming convention `<k8clustername>--default-foo.com.`
Inside this HTTPS, we will have HPP map `<K8clustername>'--default-foo.com_foo-ing1` and `<K8clustername>'--default-foo.com_bar-ing1`

Reply:
As we have multiple path and ingresses with this HTTP PS, markers will also be an array